data/lib/core/achievements.lua

```diff
@@ -0,0 +1,659 @@
+	[248] = {name = "Fire from the Earth", grade = 1, points = 2, description = "You've survived the Hellgorge eruption and found a way through the flames and lava. You've even managed to kill a few fireborn on the way."},
+	[249] = {name = "Gatherer", grade = 1, points = 2, description = "By killing creatures of the hive and gaining weapons for further missions, you started a quite effective way of war. You gathered a lot of dissolved chitin to resupply the war effort."},
+	[250] = {name = "Gem Cutter", grade = 1, points = 1, secret = true, description = "You cut your first gem - and it bears your own name! Now that would be a nice gift! This does not make it a \"true\" Heart of the Sea, however..."},	
+	[251] = {name = "Goldhunter", grade = 1, points = 2, secret = true, description = "	If it wasn't for you, several banks in Tibia would've gotten bankrupt by now. Keep on chasing bank robbers and no one will have to worry about the Tibian economy!"},	
+	[252] = {name = "Guard Killer", grade = 1, points = 2, description = "You have proven that you can beat the best of the hive. You have caused first promising breaches in the defence of the hive"},
+	[253] = {name = "Guardian Downfall", grade = 2, points = 4, description = "You ended the life of over three hundred Deepling Guards. Not quite the guardian of the Deeplings, are you?"},
+	[254] = {name = "Headache", grade = 1, points = 2, description = "Even in the deepest structures of the hive, you began to strike against the mighty foe. Your actions probably already gave the hive a headache."},	
+	[255] = {name = "Heartburn", grade = 1, points = 3, description = "Never-tiring, you attack the inner organs of the mighty hive. Your attacks on the hive's digestion system begin to cause some trouble."},
+	[256] = {name = "Hickup", grade = 1, points = 2, description = "You have grown accustomed to frequenting the hive's stomach system. Your actions have caused the hive some first digestion problems."},
+	[257] = {name = "Hive Blinder", grade = 2, points = 4, description = "You have put a lot of time and energy into keeping the hive unaware of what is happening on Quirefang. The hive learnt to fear your actions. It would surely crush you with all its might ... if it could only find you!"},	
+	[258] = {name = "Hive Fighter", grade = 1, points = 1, description = "You have participated that much in the hive war, that you are able to create some makeshift armor from the remains of dead hive born that can be found in the major hive, to show of your skill."},
+	[259] = {name = "Hive Infiltrator", grade = 1, points = 3, description = "The most powerful warriors of the hive were killed by you by the dozens. The hive is not safe anymore because of your acti
```

Comment:
Update the variable names to make it more clear what we are storing

Reply:
sorry but I don't know what they do and there are so many of them that it would be nice if someone could do a list of the changes

Reply:
Once again, github seems to be the issue, click changes and scroll down to line 472 (local t = ...)

Variables like those (in most loops) could have better names.
No an issue merging the PR but to keep the code clean and readable it should be changed.

Reply:
yeah, but my point still, I don't know what each one is storing xD

Reply:
We are trying to get info from an achievement and we need 1 variable to do that (id)
Then we loop the achievements and if we find one with the same id we create a new table (t)
We add the id and the storage and the content from achievements[id]

So to make it simple you could name them achievementsToReturn / targetAchievement or something like that.
Almost all (if not all) can be named that since the tables are for achievements.

Reply:
like this?